etcdserver/server.go

```diff
@@ -528,27 +532,20 @@ func (s *EtcdServer) run() {
 	s.r.start(s)
 
 	// asynchronously accept apply packets, dispatch progress in-order
-	ep := etcdProgress{
+	appdonec := s.startApplier(etcdProgress{
 		confState: snap.Metadata.ConfState,
 		snapi:     snap.Metadata.Index,
 		appliedi:  snap.Metadata.Index,
-	}
-	etcdprogc := s.newApplier(ep)
+	})
 
 	defer func() {
 		s.r.stop()
 		close(s.done)
-		for range etcdprogc {
-			/* wait for outstanding applys */
-		}
+		<-appdonec
 	}()
 
 	for {
```

Comment:
Can we just kill this for loop? or we can just merge startApplier loop back here?